### ISSUE TITLE
no-use-before-define false positive

### ISSUE THREAD

**Issue by Alexsey** [eslint/eslint#9335]
**Tell us about your environment**

* **ESLint Version: 4.7.1**
* **Node Version: 8.5.0**
* **npm Version: 5.4.0**

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
<details>
<summary>Configuration</summary>

```yaml
---
parserOptions:
  ecmaVersion: 8
env:
  node: true
rules:
  no-use-before-define:
    - error
    - functions: false
```

</details>

**What did you do? Please include the actual source code causing the issue.**

```js
const x = ''

function f (a = x) {
  const x = ''
}
```

**What did you expect to happen?**

No error should occurred because first `x` is from the outer scope and second `x` is from the inner scope

**What actually happened? Please include the actual, raw output from ESLint.**
```
ESLint: 'x' was used before it was defined. (no-use-before-define)
```
![screen shot 2017-09-20 at 19 37 54](https://user-images.githubusercontent.com/6392013/30655994-45b8f4a4-9e3b-11e7-9765-7878c0730871.png)

**Comment by mysticatea** [comment]
I apology for late response.

I confirmed it on [the demo](https://eslint.org/demo/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tdXNlLWJlZm9yZS1kZWZpbmU6IGVycm9yICovXHJcblxyXG5jb25zdCB4ID0gJydcclxuXHJcbmZ1bmN0aW9uIGYgKGEgPSB4KSB7XHJcbiAgY29uc3QgeCA9ICcnXHJcbn1cclxuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo4LCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19). The `x` in default parameters refers the `x` in the outer scope. It's after defined.

**Comment by mysticatea** [comment]
It seems a bug of `eslint-scope`.

**Comment by ankeetmaini** [comment]
I'd like to try my hand at this.

**Comment by mysticatea** [comment]
@ankeetmaini Oh, sorry. I have done before I see your post.

**Comment by ankeetmaini** [comment]
No worries. I didn't start it yet. 👍
